CMakeLists.txt

```diff
@@ -387,7 +387,7 @@ include_directories(SYSTEM ${LIBUV_INCLUDE_DIRS})
 find_package(Msgpack 1.0.0 REQUIRED)
 include_directories(SYSTEM ${MSGPACK_INCLUDE_DIRS})
 
-find_package(LibLUV 1.30.0 REQUIRED)
+find_package(LibLUV 1.34.1 REQUIRED)
```

Comment:
For plugins that want to use more exposed functions. See: https://github.com/jose-elias-alvarez/null-ls.nvim/pull/23

Reply:
History: That was the first try to bump the bundled version :)

(We're not using it, but plugins are, and we might? `uv.fs_mkstem`)

Reply:
> For plugins that want to use more exposed functions. See: https://github.com/jose-elias-alvarez/null-ls.nvim/pull/23

Some plugins also require ffi, but that doesn't mean nvim does. This specifies the minimum that nvim requires.

Plugins should handle validating their base requirements are satisfied or use alternatives, rather than assuming nvim is built against a certain dependency version.

Reply:
Makes sense -- we haven't done that so far, and we shouldn't start now (with a single dependency).

_If_ that is done, it should be done across the board and documented as a breaking change. (The bump to `third-party` is sufficient to make the default build use the newer version, so leaving the minimum version lower is fine.)

Reply:
> (The bump to third-party is sufficient to make the default build use the newer version, so leaving the minimum version lower is fine.)

I don't follow. ELI5? Is there a min version and a version for using more up-to-date dependencies?

Reply:
Sorry.

So the `Lib_LUV` in the _toplevel_ `CMakeLists.txt` is the minimum version required for Neovim itself to build successfully -- it tells distributions not to try to build against _earlier_ versions by erroring out early, because the basic functionality is known to be broken then.

The version in `third-party/CMakeLists.txt` is the version Neovim bundles in CI and for the _official_ releases -- this can be include nice-to-have features but is usually ignored by distributions (who build against their own versions, which is fine as long as that version is newer than the one specified in the top-level file). 

So the latter is what you want to make newer features accessible to the plugin ecosystem (in order not to fall behind distributions); the former is what you want to _force_ distributions to upgrade their own versions (which you'd only do if Neovim itself is unusable with earlier versions).

Reply:
Got it. Thanks for explaining.

Reply:
(Teaching = learning + sharing :P)